reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/tests/FeatureScopeTest.kt

```diff
@@ -156,6 +156,49 @@ class FeatureScopeTest : MockServerTest() {
         }.doWait(LONG_WAIT_MS)
     }
 
+    @Test
+    fun mustReceiveTheEvents_whenFeatureWrite_trackingConsentPendingToGranted_switched_asynchronously() {
+        // Given
+        trackingConsent = TrackingConsent.PENDING
+        testedInternalSdkCore = Datadog.initialize(
+            context = ApplicationProvider.getApplicationContext(),
+            configuration = fakeConfiguration,
+            trackingConsent = trackingConsent
+        ) as InternalSdkCore
+        testedInternalSdkCore.registerFeature(stubFeature)
+        val featureScope = testedInternalSdkCore.getFeature(fakeFeatureName)
+        checkNotNull(featureScope)
+        fakeBatchData.forEach { rawBatchEvent ->
+            val thread = Thread {
+                featureScope.withWriteContext { _, eventBatchWriter ->
+                    eventBatchWriter.write(
+                        rawBatchEvent,
+                        fakeBatchMetadata,
+                        eventType
+                    )
+                }
+            }
+            thread.start()
+            thread.join()
```

Comment:
I'm not quite clear why do we need to spin a new thread for every write request we are making, why not put `forEach` call into a single dedicated thread? By using `.join` inside every `.forEach` call you are anyway making `.withWriteContext` invocations sequential.

Reply:
I wanted to increase the potential of a collision, basically the worse case scenario.

Reply:
But that doesn't increase the chance of collision, isn't it? There is no race condition, because of the `.join` - all write requests are strictly sequential.